.github/workflows/ui.yml

```diff
@@ -4,11 +4,8 @@ on:
   push:
     branches:
       - "main"
-    paths:
-      - "ui/**"
   pull_request:
-    paths:
-      - "ui/**"
+    types: [opened, synchronize]
```

Comment:
why do specify these two types?

Reply:
So that this action runs when the pull request is opened and synchronized. In this case, synchronized means when  commits are pushed to the pr branch, the base branch is updated.

Reply:
yeah, I understand, but doesn't CI work on all types, when we do not specify them?